src/image.rs

```
@@ -18,7 +18,119 @@ use crate::transfer::open_port;
 use crate::transfer::transceive;
 use crate::transfer::SerialSpecs;
 
-pub fn list(specs: &SerialSpecs) -> Result<serde_cbor::Value, Error> {
+fn get_rc(response_body: &serde_cbor::Value) -> Option<u32> {
+    let mut rc: Option<u32> = None;
+    if let serde_cbor::Value::Map(object) = response_body {
+        for (key, val) in object.iter() {
+            match key {
+                serde_cbor::Value::Text(rc_key) if rc_key == "rc" => {
+                    if let serde_cbor::Value::Integer(parsed_rc) = val {
+                        rc = Some(*parsed_rc as u32);
+                    }
+                }
+                _ => (),
+            }
+        }
+    }
+    rc
+}
+
+fn check_answer(request_header: &NmpHdr, response_header: &NmpHdr) -> bool
+{
+    // verify sequence id
+    if response_header.seq != request_header.seq {
+        log::debug!("wrong sequence number");
+        return false;
+    }
+
+    let expected_op_type = match request_header.op {
+        NmpOp::Read => NmpOp::ReadRsp,
+        NmpOp::Write => NmpOp::WriteRsp,
+        _ => panic!("Unexpected type")
```

I think we should avoid panic, because when using it as a library, it would be better to propagate an error, as with the bail! macro from anyhow which I used elsewhere, because then the calling application can catch it and doesn't terminate.